tests/connect_redirect/connect_redirect_tests.cpp

```diff
@@ -0,0 +1,482 @@
+// Copyright (c) Microsoft Corporation
+// SPDX-License-Identifier: MIT
+
+// This module facilitates testing various connect redirect scenarios by sending traffic to both
+// local system and a remote system, both running TCP / UDP listeners.
+
+#define CATCH_CONFIG_RUNNER
+
+#include "catch_wrapper.hpp"
+#include "ebpf_udp.h"
+#include "socket_helper.h"
+
+#include <chrono>
+#include <future>
+using namespace std::chrono_literals;
+
+#include "bpf/bpf.h"
+#pragma warning(push)
+#pragma warning(disable : 4200)
+#include "bpf/libbpf.h"
+#pragma warning(pop)
+#include "catch_wrapper.hpp"
+#include "common_tests.h"
+#include "ebpf_nethooks.h"
+#include "ebpf_structs.h"
+#include "socket_helper.h"
+#include "socket_tests_common.h"
+
+#include <mstcpip.h>
+
+#define CLIENT_MESSAGE "eBPF for Windows!"
+
+static std::string _family;
+static std::string _protocol;
+static std::string _vip_v4;
+static std::string _vip_v6;
+static std::string _local_ip_v4;
+static std::string _local_ip_v6;
+static std::string _remote_ip_v4;
+static std::string _remote_ip_v6;
+static uint16_t _remote_port = 4444;
+
+typedef struct _test_addresses
+{
+    struct sockaddr_storage loopback_address;
+    struct sockaddr_storage remote_address;
+    struct sockaddr_storage local_address;
+    struct sockaddr_storage vip_address;
+} test_addresses_t;
+
+typedef struct _test_globals
+{
+    ADDRESS_FAMILY family;
+    IPPROTO protocol;
+    uint16_t remote_port;
+    test_addresses_t addresses[socket_family_t::Max];
+} test_globals_t;
+
+static test_globals_t _globals;
+static volatile bool _globals_initialized = false;
+
+inline static IPPROTO
+_get_protocol_from_string(std::string protocol)
+{
+    if (protocol.compare("udp") == 0 || protocol.compare("UDP") == 0) {
+        return IPPROTO_UDP;
+    }
+    return IPPROTO_TCP;
+}
+
+inline static ADDRESS_FAMILY
+_get_address_family_from_string(std::string family)
```

Comment:
Why not "IPv4" and "IPv6" - will be consistent with windows net powershell cmdlets.

Reply:
Actually this function is no longer being used. Removed it now.